src/version.rs

```diff
@@ -193,13 +193,9 @@ fn set_comparator(
             }
             Ok(pred)
         }
-        semver::Op::Exact => Ok(assign_partial_req(version, pred)),
-        semver::Op::Greater | semver::Op::GreaterEq | semver::Op::Less | semver::Op::LessEq => {
-            let user_pred = pred.to_string();
-            Err(unsupported_version_req(user_pred))
+        semver::Op::Exact | semver::Op::Tilde | semver::Op::Caret => {
+            Ok(assign_partial_req(version, pred))
         }
-        semver::Op::Tilde => Ok(assign_partial_req(version, pred)),
-        semver::Op::Caret => Ok(assign_partial_req(version, pred)),
```

Comment:
This was actually intentional.   I'm wanting to treat it like its exhaustive so you can see what our intended policy is but it isn't exhaustive, so we have to have the `_`